librz/core/cmd.c

```diff
@@ -6183,6 +6183,7 @@ static RzCmdStatus core_cmd_tsr2cmd(RzCore *core, const char *cstr, bool split_l
 	ts_parser_set_language(parser, (TSLanguage *)core->rcmd->language);
 
 	TSTree *tree = ts_parser_parse_string(parser, NULL, input, strlen(input));
+	rz_return_val_if_fail(tree, RZ_CMD_STATUS_INVALID);
```

Comment:
shouldn't this be an if instead of that ?

Reply:
it seems this happen only if something completely wrong is going on (e.g. in this case because the version used to generate the tree-sitter parser was incompatible with the one linked with rizin). However you are right as here it's probably better to have a `if`, but I will place a `rz_return_val_if_fail` on the return value of the call to `ts_parser_set_language`, because when that returns false there was a version mismatch and that's an error we might consider as a programmer's one and where i'd like to fail loudly and early.